src/Data/Sequences.hs

```diff
@@ -411,6 +410,12 @@ class (Monoid seq, MonoTraversable seq, SemiSequence seq, MonoPointed seq) => Is
     unsafeIndex :: seq -> Index seq -> Element seq
     unsafeIndex = indexEx
 
+
+    -- | 'splitWhen' splits a sequence into components delimited by separators, where the
+    -- predicate returns True for a separator element.
```

Comment:
Would you mind adding a comment about whether the separators are retained or discarded? A concrete example would be useful. At least I personally get confused about this whenever I use split-style functions.